src/storybook/utils.ts

```diff
@@ -0,0 +1,68 @@
+import {ContrastColorFs, ThemeFs} from '@atb-as/theme';
+import {ContrastColor, themes} from '@atb/theme/colors';
+import {isDefined} from '@atb/utils/presence';
+
+/**
+ * Recursively adds a key prefix to a set of contast colors, and returns the
+ * flattened object.
+ *
+ * @example
+ * let backgrounds = flattenColorsWithPrefix('background', {
+ *   vibrant: {background: '#ff0', foreground: {...}},
+ *   neutral: {
+ *     0: {background: '#fff', foreground: {...}},
+ *   },
+ * });
+ * // returns
+ * {
+ *   'background vibrant': { background: '#ff0', foreground: {...} }
+ *   'background neutral 0': { background: '#fff', foreground: {...} }
+ * }
+ */
+function flattenColorsWithPrefix(
+  prefix: string,
+  options: object,
```

Comment:
It's built to support any object that has ContrastColors as leaf nodes, and since the function is recursive, `Theme['color']` only applies to the first step.

I followed your example, and asked ChatGPT about it, and I think `Nested<ContrastColor>` will work here 😄 Where Nested is:

```ts
type Nested<T> = T | {[key: string]: Nested<T>};
```

Reply:
Would be a nice improvement if that works!

Reply:
Awesome! 🥳

Reply:
Seems to work great 🕺 

```ts
const test: Nested<ContrastColorFs> = {
  colorset: {
    color: themes.light.color.background.accent, // this works!
  },
  foo: 'bar', // this gives a ts error
};
```